core/schema/module.go

```diff
@@ -836,16 +841,48 @@ func (s *moduleSchema) moduleInitialize(
 	inst dagql.Instance[*core.Module],
 	args struct{},
 ) (*core.Module, error) {
-	if inst.Self.NameField == "" || inst.Self.SDKConfig == nil || inst.Self.SDKConfig.Source == "" {
+	// If SDK config is set, we expect the name and sdk source field to be set
+	if inst.Self.SDKConfig != nil && (inst.Self.NameField == ""  || inst.Self.SDKConfig.Source == "") {
 		return nil, fmt.Errorf("module name and SDK must be set")
 	}
+
 	mod, err := inst.Self.Initialize(ctx, inst.ID(), dagql.CurrentID(ctx), s.dag)
 	if err != nil {
 		return nil, fmt.Errorf("failed to initialize module: %w", err)
 	}
 	return mod, nil
 }
 
+func (s *moduleSchema) moduleGenerateClient(
+	ctx context.Context,
+	mod *core.Module,
+	args struct {
+		Generator dagql.String
+		LocalSdk dagql.Boolean
```

Comment:
This should be an optional?

Reply:
You're right! It's a missing fix while I was debugging the schema generation that was broken